### PR TITLE
Start supporting arm32v7 armv7 starting with 3.9

### PR DESCRIPTION
**Pull request by oliv3r**
With alpine 3.9 out the door, and it now officially supporting armv7 (http://dl-cdn.alpinelinux.org/alpine/v3.9/main/armv7/) also start creating docker builds for it.

### CODE REVIEW

**Comment by tianon** [comment]
The "alpine:3.9" Docker image doesn't support arm32v7 yet.

**Comment by oliv3r** [comment]
@tianon what did you mean with your comment? I know the docker images do not support it yet; but alpine now has armv7 builds out, so this should be the first step to generate them right?

**Comment by tianon** [comment]
Doh sorry yeah, I misread which repo this was on 😅

In this case, the RHS of this case should be `arm32v7` to match official images arch designations.

**Comment by oliv3r** [comment]
> In this case, the RHS of this case should be `arm32v7` to match official images arch designations.

Not sure what you mean here either :) What does RHS stand for? Is there anything that's expected here from me?

File: build [hunk 1]
```diff
@@ -122,7 +122,8 @@ library() {
 			case "$arch" in # converts Alpine Linux arch strings to Docker arch strings
 				x86_64) echo -n "amd64";;
 				x86) echo -n "i386";;
-				armhf) echo -n "arm32v6";; # arm32v7 is not officially supported by Alpine, but arm32v6 should still work.
+				armhf) echo -n "arm32v6";;
+				armv7) echo -n "armv7";;
```
Comment (tianon) [inline build]: Sorry, by RHS, I meant Right-Hand-Side:

```suggestion
				armv7) echo -n "arm32v7";;
```

Comment (tianon) [reply build]: It's also worth noting that I'm not sure if this change is all that's required to add another architecture here -- I think there are more files that need to change for everything to work.

@ncopa @andyshinn any comment/suggestion?

I think it's also worth noting that 3.9 doesn't appear to have a 3.9 "minirootfs" artifact for `armv7`: http://dl-cdn.alpinelinux.org/alpine/v3.9/releases/armv7/

(Compare that to `armhf`: http://dl-cdn.alpinelinux.org/alpine/v3.9/releases/armhf/)

Comment (oliv3r) [reply build]: Yeah, I think it needs: https://github.com/oliv3r/aports/tree/armv7 to actually generate the images.
Anyway, most important part is to get the ball rolling right ;)

**Comment by frenck** [comment]
Superseded by #487 ?

**Comment by oliv3r** [comment]
> Superseded by #487 ?

I would rather say #487 is a duplicate?

Anyway, I created the changes I thought where needed for armv7 images last week; and they have since been merged.

So I think we now need to check how useable these images are, if there was anything ommitted in the alpine armv7 PR and what was missing in this PR to get these images going?

**Comment by oliv3r** [comment]
I saw that there's no script that generates the armv7 subdirectory, even though it seems to be completely identical to a least the armhf variant.

What I am not clear on, is how the versioning works. Currently only alpine 3.9.1 and 3.9.2 minirootfs are built, but the option files list 3.9.0. I'll open a ticket on alpine to see if they can back-generate the 3.9.0 rootfs for completness.

**Comment by oliv3r** [comment]
Also, the 'supported architectures' on docker hub is probably done at a later point in the official repo, I suppose only maintainers can do that later, correct?

**Comment by frenck** [comment]
> I would rather say #487 is a duplicate?

Fine by me, at the time of writing, #487 was more complete, hence my comment.
Personally, I'm desperately looking forward to using armv7, whose PR is getting merged is not a concern to me.

**Comment by oliv3r** [comment]
So I can build local armv7 containers, so this PR is 'complete' but requires a version bump to 3.9.1 or 3.9.2 as there is no armv7 3.9.0 over on alpine. Not sure if the 3.9.0 bump for armv7 is heared there though (alpinelinux/aports#6444). So I'll probably hit @ncopa up on IRC.

**Comment by oliv3r** [comment]
We best close this pull-request as the organizational structure has changed, with alpine linux now hosting their own Dockerfiles. Docker hub now seems to have armv7 so we should be able to use those as well.

**Pull request closed without merging**